src/main/java/com/meta/chatbridge/llm/HuggingFaceLlamaPlugin.java

```diff
@@ -0,0 +1,187 @@
+/*
+ *
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package com.meta.chatbridge.llm;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.knuddels.jtokkit.Encodings;
+import com.knuddels.jtokkit.api.Encoding;
+import com.meta.chatbridge.message.Message;
+import com.meta.chatbridge.message.MessageStack;
+import kotlin.Pair;
+import org.apache.hc.client5.http.fluent.Request;
+import org.apache.hc.client5.http.fluent.Response;
+import org.apache.hc.core5.http.ContentType;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.common.returnsreceiver.qual.This;
+import org.jetbrains.annotations.TestOnly;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class HuggingFaceLlamaPlugin<T extends Message> implements LLMPlugin<T> {
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+    private final HuggingFaceConfig config;
+    private URI endpoint;
+
+    public HuggingFaceLlamaPlugin(HuggingFaceConfig config) {
+        this.config = config;
+
+        try {
+            this.endpoint = new URI(this.config.endpoint());
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e); // this should be impossible
```

Comment:
this is totally possible :)

Reply:
Changed in the config to double check that it's a URI

Reply:
well my point here is if we're checking it's a URI why not make it a uri so that config.endpoint() returns a URI.